src/components/WalletButton/WalletButton.tsx

```diff
@@ -29,37 +47,111 @@ export type WalletButtonProps = {
    * Callback function for when disconnect button is clicked
    */
   onDisconnectWalletClicked: () => void;
+  /**
+   * List of transactions that have been submitted
+   */
+  transactions: SubmittedTransaction[];
+  /**
+   * Number representing chainId of ETH network
+   */
+  chainId: number;
+  /**
+   * List of all tokens in metadata
+   */
+  tokens: TokenInfo[];
 };
 
 export const WalletButton = ({
   address,
-  className,
   isConnecting,
   onConnectWalletClicked,
   onDisconnectWalletClicked,
+  transactions,
+  chainId,
+  tokens,
 }: WalletButtonProps) => {
   const { t } = useTranslation(["wallet"]);
+
+  const [walletOpen, setWalletOpen] = useState<boolean>(false);
+
   return (
-    <div className={className}>
+    <Container>
       {address ? (
-        <StyledWalletButton>
-          <Blockies
-            size={8}
-            scale={3}
-            seed={address}
-            className="-ml-1 rounded-sm mr-2"
-            bgColor="black"
-            color="#2b72ff"
-          />
-          <span>{truncateEthAddress(address)}</span>
-          <button
-            className="ml-2"
-            aria-label={t("wallet:disconnectWallet")}
-            onClick={onDisconnectWalletClicked}
-          >
-            <RiCloseLine />
-          </button>
-        </StyledWalletButton>
+        <>
+          <StyledWalletButton onClick={() => setWalletOpen(!walletOpen)}>
+            <BlockiesContainer>
+              <StyledBlockies
+                size={8}
+                scale={3}
+                seed={address}
+                bgColor="black"
+                color="#2b72ff"
+              />
+              <GreenCircle />
+            </BlockiesContainer>
+            <InfoHeading>{truncateEthAddress(address)}</InfoHeading>
+            {walletOpen && (
+              <ExitButton onClick={() => setWalletOpen(!walletOpen)}>
+                <RiCloseLine />
+              </ExitButton>
+            )}
+          </StyledWalletButton>
+          {walletOpen && (
+            <WalletExtension>
+              <Line />
+              <TransactionContainer flex={transactions.length === 0}>
+                {transactions.length > 0 ? (
```

Comment:
Remove this line, if you slice an empty array it's still empty. `[].slice(0, 3);`

Reply:
This line is left so that we can render "no transaction" if transactions is empty.

Reply:
Ah yes I didn't notice.